op2/src/common/op_perf_common.cpp

```diff
@@ -0,0 +1,164 @@
+
+/*******************************************************************************
+ * Routine to linear search comm_info array in an op_comm_kernel for an op_dat
+ *******************************************************************************/
+int search_op_comm_kernel(op_dat dat, op_comm_kernel *kernal, int num_indices) {
+  for (int i = 0; i < num_indices; i++) {
+    if (strcmp((kernal->comm_info[i])->name, dat->name) == 0 &&
+        (kernal->comm_info[i])->size == dat->size) {
+      return i;
+    }
+  }
+
+  return -1;
+}
+
+/*******************************************************************************
+ * Routine to measure message sizes exchanged in an op_par_loop / kernel
+ *******************************************************************************/
+void op__perf_comm(void *k_i, op_dat dat) {
+  halo_list exp_exec_list = OP_export_exec_list[dat->set->index];
+  halo_list exp_nonexec_list = OP_export_nonexec_list[dat->set->index];
+  int tot_halo_size =
+      (exp_exec_list->size + exp_nonexec_list->size) * (size_t)dat->size;
+
+  op_comm_kernel *kernel_entry = (op_comm_kernel *)k_i;
+  int num_indices = kernel_entry->num_indices;
+
+  if (num_indices == 0) {
+    // set capcity of comm_info array
+    kernel_entry->cap = 20;
+    op_dat_mpi_comm_info dat_comm =
+        (op_dat_mpi_comm_info)xmalloc(sizeof(op_dat_mpi_comm_info_core));
+    kernel_entry->comm_info = (op_dat_mpi_comm_info *)xmalloc(
+        sizeof(op_dat_mpi_comm_info *) * (kernel_entry->cap));
+    strncpy((char *)dat_comm->name, dat->name, 20);
+    dat_comm->size = dat->size;
+    dat_comm->index = dat->index;
+    dat_comm->count = 0;
+    dat_comm->bytes = 0;
+
+    // add first values
+    dat_comm->count += 1;
+    dat_comm->bytes += tot_halo_size;
+
+    kernel_entry->comm_info[num_indices] = dat_comm;
+    kernel_entry->num_indices++;
+  } else {
+    int index = search_op_comm_kernel(dat, kernel_entry, num_indices);
+    if (index < 0) {
+      // increase capacity of comm_info array
+      if (num_indices >= kernel_entry->cap) {
+        kernel_entry->cap = kernel_entry->cap * 2;
+        kernel_entry->comm_info = (op_dat_mpi_comm_info *)xrealloc(
+            kernel_entry->comm_info,
+            sizeof(op_dat_mpi_comm_info *) * (kernel_entry->cap));
+      }
+
+      op_dat_mpi_comm_info dat_comm =
+          (op_dat_mpi_comm_info)xmalloc(sizeof(op_dat_mpi_comm_info_core));
+
+      strncpy((char *)dat_comm->name, dat->name, 20);
+      dat_comm->size = dat->size;
+      dat_comm->index = dat->index;
+      dat_comm->count = 0;
+      dat_comm->bytes = 0;
+
+      // add first values
+      dat_comm->count += 1;
+      dat_comm->bytes += tot_halo_size;
+
+      kernel_entry->comm_info[num_indices] = dat_comm;
+      kernel_entry->num_indices++;
+    } else {
+      kernel_entry->comm_info[index]->count += 1;
+      kernel_entry->comm_info[index]->bytes += tot_halo_size;
+    }
+  }
+}
+#endif
+
+#ifdef COMM_PERF
```

Comment:
These lines are unnecessary, it's the same #ifdef as above.

Reply:
Yeah, I know, but it's how it was done in the original file. I could have changed it, but there might be a reason for it.